packages/inferno/src/DOM/unmounting.ts

```diff
@@ -28,11 +28,17 @@ export function unmount(vNode) {
 
     if (!isNull(props)) {
       const keys = Object.keys(props);
+      const doc = vNode.dom.ownerDocument || document;
```

Comment:
delegatedEvents[key] is used for detecting semi synthetic event. Doc ID and all this code should not be needed. We want to keep this loop as fast as possible

Reply:
doc is now just passed as argument like in other places